### PR TITLE
Fix easy bits of Datastore API review

### PR DESCRIPTION
**Pull request by jskeet**
Trickier bits around query handling still TBD.

(Worth reviewing one commit at a time.)

### CODE REVIEW

**Comment by mlmcmullen** [comment]
For the renaming of `CreateInsertionKey` to `CreateIncompleteKey`, it is also referenced in some example code in the [README.md](https://github.com/GoogleCloudPlatform/gcloud-dotnet/blob/master/README.md#example-1) which would probably be good to update at the same time.

**Comment by jskeet** [comment]
@mlmcmullen: Well spotted. Will fix now.

File: snippets/Google.Datastore.V1Beta3.Snippets/DatastoreDbSnippets.cs [hunk 1]
```diff
@@ -660,7 +660,7 @@ public void GroupingQuery()
                 DistinctOn = { "type" },
                 Order =
                 {
-                    { "type", Direction.Ascending },
+                    "type", // If only the name is specified, it's implicitly ascending
                     { "priority", Direction.Ascending }
```
Comment (chrisdunelm) [inline snippets/Google.Datastore.V1Beta3.Snippets/DatastoreDbSnippets.cs]: You've left the "Direction.Ascending" here to show how to specify direction explicitly?

File: src/Google.Datastore.V1Beta3/QueryExtensions.cs [hunk 2]
```diff
@@ -64,5 +64,16 @@ public static void Add(this RepeatedField<PropertyOrder> orderings, string prope
                 nameof(direction), "Direction must be Ascending or Descending");
```
Comment (chrisdunelm) [inline src/Google.Datastore.V1Beta3/QueryExtensions.cs]: I know it's not relevant to this change, but why doesn't this CheckArgument use the enum checker?

**Comment by chrisdunelm** [comment]
LGTM

File: snippets/Google.Datastore.V1Beta3.Snippets/DatastoreDbSnippets.cs (hunk 1 above)
Comment (jskeet) [inline snippets/Google.Datastore.V1Beta3.Snippets/DatastoreDbSnippets.cs]: I think so :) I would make the second descending, but that wouldn't match the API documentation example.
Maybe I should add a comment saying "// Either direction can be specified explicitly" on the second line?

Comment (chrisdunelm) [inline snippets/Google.Datastore.V1Beta3.Snippets/DatastoreDbSnippets.cs]: Yes I think that'd be useful :)

File: src/Google.Datastore.V1Beta3/QueryExtensions.cs (hunk 2 above)
Comment (chrisdunelm) [inline src/Google.Datastore.V1Beta3/QueryExtensions.cs]: :) Fair enough!
Sounds like something that may be worth doing later.

File: snippets/Google.Datastore.V1Beta3.Snippets/DatastoreDbSnippets.cs (hunk 1 above)
Comment (jskeet) [inline snippets/Google.Datastore.V1Beta3.Snippets/DatastoreDbSnippets.cs]: Done. Will rebase and commit.

**Pull request merged**